pkg/pidhandle/pidhandle.go

```diff
@@ -0,0 +1,134 @@
+//go:build !windows
+
+// Package for handling processes and PIDs.
+package pidhandle
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/shirou/gopsutil/v4/process"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+// PIDHandle defines an interface for working with operating system processes
+// in a reliable way. OS-specific implementations include additional logic
+// to try to ensure that operations (e.g., sending signals) are performed
+// on the exact same process that was originally referenced when the PIDHandle
+// was created via NewPIDHandle or NewPIDHandleFromString.
+//
+// This prevents accidental interaction with a different process in scenarios
+// where the original process has exited and its PID has been reused by
+// the system for an unrelated process.
+type PIDHandle interface {
+	// Returns the PID associated with this PIDHandle.
+	PID() int
+	// Releases the PIDHandle resources.
+	Close() error
+	// Sends the signal to process.
+	Kill(signal unix.Signal) error
+	// Returns true in case the process is still alive.
+	IsAlive() (bool, error)
+	// Returns a serialized representation of the PIDHandle.
+	// This string can be passed to NewPIDHandleFromString to recreate
+	// a PIDHandle that reliably refers to the same process as the original.
+	String() (string, error)
+}
+
+const noSuchProcessID = "no-proc"
```

Comment:
the `name-to-handle` and `start-time` should likely be defined here as well together with some basic comment what they represent.